config/api_client.go

```diff
@@ -34,8 +34,8 @@ func (c *Config) NewApiClient() (*httpclient.ApiClient, error) {
 		DebugTruncateBytes: c.DebugTruncateBytes,
 		InsecureSkipVerify: c.InsecureSkipVerify,
 		Transport:          c.HTTPTransport,
+		AuthVisitor:        c.Authenticate,
 		Visitors: []httpclient.RequestVisitor{
-			c.Authenticate,
```

Comment:
why duplication? AuthVisitor is still a visitor.

Reply:
We will be overriding auth per method for direct-to-dataplane operations. These will not use the cached header factory, they'll use a new one that is injected on a per-method basis depending on the endpoint in question.